src/payment/strategies/clearpay/clearpay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,250 @@
+
+            strategy.execute(payload).catch(errorHandler);
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(errorHandler).toHaveBeenCalled();
+        });
+
+        it('throws error if trying to execute before initialization', async () => {
+            await strategy.deinitialize();
+
+            try {
+                await strategy.execute(payload);
+            } catch (error) {
+                expect(error).toBeInstanceOf(NotInitializedError);
+            }
+        });
+
+        it('loads payment client token', () => {
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalledWith(paymentMethod.gateway, undefined);
+            expect(store.dispatch).toHaveBeenCalledWith(loadPaymentMethodAction);
+        });
+    });
+
+    describe('#finalize()', () => {
+        const nonce = 'bar';
+
+        it('submits the order and the payment', async () => {
+            store = createCheckoutStore(merge({}, getCheckoutStoreState(), {
+                config: {
+                    data: {
+                        context: { payment: { token: nonce } },
+                    },
+                },
+                checkout: {
+                    data: {
+                        ...getCheckout(),
+                        payments: [{
+                            ...getCheckoutPayment(),
+                            providerId: paymentMethod.id,
+                            gatewayId: paymentMethod.gateway,
+                        }],
+                    },
+                },
+                order: {},
+            }));
+
+            strategy = new ClearpayPaymentStrategy(
+                store,
+                checkoutValidator,
+                orderActionCreator,
+                paymentActionCreator,
+                paymentMethodActionCreator,
+                storeCreditActionCreator,
+                scriptLoader
+            );
+
+            jest.spyOn(store, 'dispatch');
+
+            await strategy.initialize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
+            await strategy.finalize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
+
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(
+                {},
+                { methodId: paymentMethod.id, gatewayId: paymentMethod.gateway }
+            );
+
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith({
+                methodId: paymentMethod.id,
+                paymentData: { nonce },
+            });
+        });
+
+        it('throws error if unable to finalize order due to missing data', async () => {
+            try {
+                await strategy.finalize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
```

Comment:
for testing exceptions, the following approach is preferred as it ensures the assertion is executed:
```js
expect(() => await strategy.finalize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway)
  .toThrow(MissingDataError)
```